WooCommerce/Classes/ViewRelated/Inbox/InboxNoteRow.swift

```diff
@@ -53,29 +57,74 @@ struct InboxNoteRow: View {
                                 Text(action.title)
                             }
                         }
-                        Button(Localization.dismiss) {
-                            // TODO: 5955 - handle dismiss action
-                            print("Handling dismiss action")
+                        if dismissButtonIsLoading {
+                            ActivityIndicator(isAnimating: .constant(true), style: .medium)
                         }
+                        else {
+                            Button(Localization.dismiss) {
+                                dismissButtonIsLoading = true
+                                viewModel.dismissInboxNote { _ in
+                                    dismissButtonIsLoading = false
+                                }
+                            }
                         .foregroundColor(Color(.withColorStudio(.gray, shade: .shade30)))
                         .font(.body)
                         .buttonStyle(PlainButtonStyle())
-
+                        }
                         Spacer()
                     }
                 }
             }
-                   .padding(Constants.defaultPadding)
+            .padding(Constants.defaultPadding)
+            .sheet(isPresented: $showWebView, content: {
+                webView
+            })
 
             Divider()
                 .frame(height: Constants.dividerHeight)
         }
     }
+
+    @ViewBuilder
+    private var webView: some View {
+        let isWPComStore = ServiceLocator.stores.sessionManager.defaultSite?.isWordPressStore ?? false
+
+        if isWPComStore {
+        NavigationView {
+            AuthenticatedWebView(isPresented: $showWebView,
+                                 url: displayedURL?.absoluteURL ?? WooConstants.URLs.blog.asURL(),
+                                 urlToTriggerExit: nil) {
+
+            }
+             .navigationTitle(Localization.inboxWebViewTitle)
+             .navigationBarTitleDisplayMode(.inline)
+             .toolbar {
+                 ToolbarItem(placement: .confirmationAction) {
+                     Button(action: {
+                         showWebView = false
+                     }, label: {
+                         Text(Localization.doneButtonWebview)
+                     })
+                 }
+             }
+        }
+        .wooNavigationBarStyle()
+        }
+        else {
+            SafariSheetView(url: displayedURL ?? WooConstants.URLs.blog.asURL())
```

Comment:
nit: does it make sense to `guard let` the URL and show an empty view or make a non-optional URL at the beginning of the function to be reused in both cases?
```suggestion
        let isWPComStore = ServiceLocator.stores.sessionManager.defaultSite?.isWordPressStore ?? false
        let url = displayedURL?.absoluteURL ?? WooConstants.URLs.blog.asURL()

        if isWPComStore {
        NavigationView {
            AuthenticatedWebView(isPresented: $showWebView,
                                 url: url,
                                 urlToTriggerExit: nil) {

            }
             .navigationTitle(Localization.inboxWebViewTitle)
             .navigationBarTitleDisplayMode(.inline)
             .toolbar {
                 ToolbarItem(placement: .confirmationAction) {
                     Button(action: {
                         showWebView = false
                     }, label: {
                         Text(Localization.doneButtonWebview)
                     })
                 }
             }
        }
        .wooNavigationBarStyle()
        }
        else {
            SafariSheetView(url: url)
```